src/chain/chain_state.cpp

```diff
@@ -317,8 +321,14 @@ uint32_t chain_state::work_required_retarget(const data& values,
     BITCOIN_ASSERT_MSG(!bits.is_overflowed(), "previous block has bad bits");
 
     uint256_t target(bits);
+    const bool needs_shift = msb(target) > msb(pow_limit) - 1;
```

Comment:
This is problematic now that it is parameterized:
```cpp
static const uint256_t pow_limit(compact{ retarget_proof_of_work_limit });
```
The `retarget_proof_of_work_limit` parameter can change but the result will not.